core/src/transport.rs

```diff
@@ -333,6 +333,12 @@ pub enum TransportError<TErr> {
     Other(TErr),
 }
 
+impl<TErr> From<TErr> for TransportError<TErr> where TErr: std::error::Error {
```

Comment:
Unfortunately this will make it impossible to implement `From<Something> for TransportError` in the future if we happen to add other variants to `TransportError`, so we should remove this.